### PR TITLE
[dev] Refactor validation module

### PR DESCRIPTION
**Pull request by m-kus**
- [x] issue #91 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/raito/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

### CODE REVIEW

**Review by TAdev0 — APPROVED** [review]
nice! lgtm!

File: src/state.cairo [hunk 1]
```diff
@@ -185,3 +191,84 @@ impl TxOutDefault of Default<TxOut> {
         TxOut { value: 0, pk_script: @"", cached: false, }
     }
 }
+
+#[generate_trait]
+pub impl TransactionImpl of TransactionTrait {
```
Comment (TAdev0) [inline src/state.cairo:196]: why did you choose to place `TransactionImpl` and `HeaderImpl` in state.cairo? just by curiosity.

File: src/codec.cairo [hunk 2]
```diff
@@ -38,4 +40,57 @@ pub fn encode_compact_size(value: usize, ref dest: ByteArray) { // TODO:
 }
 // TODO: implement Encode for Transaction, TxIn, TxOut, OutPoint
 
+/// Encodes transaction (TODO: remove me, use Encode instead)
+pub fn encode_transaction(tx: @Transaction, _segwit: bool) -> ByteArray {
```
[the remark this answers is not in the document]
Comment (m-kus) [reply src/codec.cairo:44]: It's just not used yet, thus the prefix

File: src/state.cairo (hunk 1 above)
Comment (m-kus) [reply src/state.cairo:196]: That's to be addressed in the next PR :)
But generally it's easier to look up the traits implemented for a particular type if they are in the same file

**Review by lana-shanghai — APPROVED** [review]
lgtm

**Pull request merged**